### PR TITLE
(PUP-7042) Mark strings in type

### PR DESCRIPTION
**Pull request by Magisus**
This commit marks user-facing error and info strings in
`lib/puppet/type/*` for translation.

### CODE REVIEW

**Comment by Magisus** [comment]
Closing and reopening to fix Travis.

**Comment by Magisus** [comment]
Closing and reopening to fix CI.

File: lib/puppet/type/macauthorization.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def munge_boolean(value)
   def munge_integer(value)
       Integer(value)
   rescue ArgumentError
-      fail("munge_integer only takes integers")
+      fail "munge_integer only takes integers"
```
Comment (MosesMendoza) [inline lib/puppet/type/macauthorization.rb]: Looks like this was modified incorrectly

File: lib/puppet/type/tidy.rb [hunk 2]
```diff
@@ -258,7 +260,7 @@ def generate
     end
     found_files = files.find_all { |path| tidy?(path) }.collect { |path| mkfile(path) }
     result = found_files.each { |file| debug "Tidying #{file.ref}" }.sort { |a,b| b[:path] <=> a[:path] }
-    notice "Tidying #{found_files.size} files"
+    notice _("Tidying %{count} files") % { count: found_files.size }
```
Comment (MosesMendoza) [inline lib/puppet/type/tidy.rb]: Probably worth including note here too re: `Tidying`

File: lib/puppet/type/tidy.rb [hunk 3]
```diff
@@ -322,10 +324,10 @@ def stat(path)
     begin
       Puppet::FileSystem.lstat(path)
     rescue Errno::ENOENT => error
-      info "File does not exist"
+      info _("File does not exist")
       return nil
     rescue Errno::EACCES => error
-      warning "Could not stat; permission denied"
+      warning _("Could not stat; permission denied")
```
Comment (MosesMendoza) [inline lib/puppet/type/tidy.rb]: Could you add a note re: `stat` is a program name?

File: lib/puppet/type/user.rb [hunk 4]
```diff
@@ -391,7 +391,7 @@ def insync?(current)
 
       validate do |value|
         if value.intern != :absent and value !~ /^\d{4}-\d{2}-\d{2}$/
-          raise ArgumentError, "Expiry dates must be YYYY-MM-DD or the string \"absent\""
+          raise ArgumentError, _("Expiry dates must be YYYY-MM-DD or the string \"absent\"")
```
Comment (MosesMendoza) [inline lib/puppet/type/user.rb]: Do you think its worth a translator note that YYYY-MM-DD reflects a four digit year two digit month two digit day separated by dashes? Maybe that's self-evident, not sure how/if it translates though

File: lib/puppet/type/user.rb [hunk 5]
```diff
@@ -681,14 +681,14 @@ def generate
...
-          raise ArgumentError, "purge_ssh_keys can only be true for users with a defined home directory"
+          raise ArgumentError, _("purge_ssh_keys can only be true for users with a defined home directory")
         end
 
         return [ "#{home}/.ssh/authorized_keys" ] if value == :true
         # value is an array - munge each value
         [ value ].flatten.map do |entry|
           if entry =~ /^~|^%h/ and not home
-            raise ArgumentError, "purge_ssh_keys value '#{value}' meta character ~ or %h only allowed for users with a defined home directory"
+            raise ArgumentError, "purge_ssh_keys value '#{value}' meta character ~ or \%h only allowed for users with a defined home directory"
```
Comment (MosesMendoza) [inline lib/puppet/type/user.rb]: Looks like this line got skipped, maybe as a result of escaping `%h` ?

**Comment by Magisus** [comment]
Updated.

**Pull request merged**